DR: multi-repo setup

AlexanderLanin opened the pull request:
Dear bazel and docs-as-code people: please review the entire DR.

All other stakeholders (e.g. process experts) should focus on reviewing the use cases / requirements and constraints.

Closes #241

hoe-jo commented on docs/_tooling/decision-records/multirepo.md:
```diff
@@ -0,0 +1,116 @@
+   * Used by each module individually
+   * Unidirectional links from module to score are sufficient
+   * With metrics for feature-requirements coverage
+   * score website itself shall remain unchanged (no links to modules)
```
Sorry don´t understand your point? What do you mean with that?

AlexanderLanin replied:
The score website (feature requirements) shall not contain links to the modules (feature implementation)?!

danwos commented on docs/_tooling/decision-records/multirepo.md:
```diff
@@ -0,0 +1,116 @@
+* Good performance (no need to build all repositories at once)
+Con:
+* needs.json and the entire website needs to be hosted in multiple versions.
+* bidirectional links affect the "core score" repository, which is not desired.
```
I don't understand this point by 100%: What exactly is affected in the "core score" repository?
From my understanding changes in the repository itself are not needed, just the final documentation gets incoming links from module-repos/docs.

AlexanderLanin replied:
Yes, I was talking about the website, not the repository. Fixed:
> bidirectional links affect the score-website, which is not desired

AlexanderLanin replied:
Resolving, as the text is completely rewrtten.

danwos commented on docs/_tooling/decision-records/multirepo.md:
```diff
@@ -0,0 +1,116 @@
+
+Con:
+* Relies heavily on bazel -> potential problems with esbonio etc
+* Performance / runtime
```
This problem can be huge, depending on the content of the docs.
For instance, if each module-repo needs a doxygen/doxysphinx beuild for API -documentation.

It may be okay for a CI build, but giving the user feedback during editing (esbonio build) would take too long.

Maybe a 2-way approach/support would be good:
* A "complete" build for the CI and the deployed documentation
* A local quick & dirty build for the specific repo only, in which the current change gets implemented (for quick user feedback of ~95% of our checks/rules.)

AlexanderLanin replied:
Resolving, as the text is completely rewrtten.

danwos commented on docs/_tooling/decision-records/multirepo.md:
```diff
@@ -0,0 +1,116 @@
+
+These are the identified alternatives. We'll dive into each one in detail.
+1) weblinks
+2) needservice
```
1+2 should be exchanged, as weblinks is `2)`` in the following chapters.

danwos reviewed (commented):
Thanks for the great write-up.

I don't have strong opinion about the different solutions.

Regarding the proposed solution "Copying", I just asked myself, how this works in the different scenarios for local builds:

* `SCORE main` gets updated, most module-repos are not checked out
* Single module-repo gets updated, `SCORE main` may be locally outdated
* `SCORE main` and 1 module-repo gets updated (new feature docs + implementation)
  * Finally these changes get committed by 2 PRs. The `SCORE main` PR needs to be merged first, so that the module-PR can use a valid feature as link target. Doable, and maybe not a common use case.

For fast, local feedback it should be possible to build the docs with `SCORE main` and one selected module-repo only.
Do you think this is possible? I'm a bazel newbie, so I'm unsure :)

MaximilianSoerenPollak commented on docs/_tooling/decision-records/multirepo.md:
```diff
@@ -162,46 +125,87 @@ Con:
+- Con: No customizations via conf.py possible, e.g. module specific templates.
 
-### 6) Needs-external-needs (+ intersphinx)
+Apprach 2 (separate websites):
```
```suggestion
Approach 2 (separate websites):
```

AlexanderLanin replied:
As long as that's the only problem in the document, I'm quite happy 🤣

MaximilianSoerenPollak commented on docs/_tooling/decision-records/multirepo.md:
```diff
@@ -0,0 +1,252 @@
+
+After that is implemented, we'll evaluate whether we can switch to
+bidirectional linking. Especially for the integration-build. We'll use solution
+1 or 3 implement the solution mentioned in 'Further Thoughts'. In any way, by
```
Is this `1, 3 or implement the solution...` 

Seems like something is missing here?

MaximilianSoerenPollak reviewed (approved):
LGTM, just one comment regarding sentence understandability.

AlexanderLanin commented:
Please have a look whether option 2 still is our favorite!
I've rewritten option 3 and added a summary.


@PhilipPartsch @MaximilianSoerenPollak @ltekieli @nradakovic @hoe-jo @danwos
